swimlane/core/cursor.py

```diff
@@ -29,36 +29,49 @@ class PaginatedCursor(Cursor):
 
     default_limit = 0
     default_page_size = 10
+    pstart = -1
+    pend = -1
 
-    def __init__(self, limit=default_limit, page_size=default_page_size):
+    def __init__(self, limit=default_limit, page_size=default_page_size, page_start=pstart, page_end=pend):
         super(PaginatedCursor, self).__init__()
 
         self.__limit = limit
         self.page_size = page_size
+        self.page_start = page_start
+        self.page_end = self.page_start + 1 if (page_end == -1 or page_end <= page_start) else page_end
 
         if self.__limit:
             self.page_size = min(self.page_size, self.__limit)
 
+
     def _evaluate(self):
         """Lazily retrieve and paginate report results and build Record instances from returned data"""
         if self._elements:
             for element in self._elements:
                 yield element
         else:
-            for page in itertools.count():
+            # Determine pagination range based on parameters
+            if self.__limit == 0 and self.page_start != -1:
```

Comment:
This will break for query,
```python
records.search(
    limit=0
    page_start = 10
)
``` 
Here we can still call itertools with count param as page_start
`itertools.count(self.page_start)`

The ideal condition is, page_end != -1. And as mentioned in previous command, raise exception for scenario where page_end > 0  but page_start = None